components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -1168,33 +1168,25 @@ public static String getOAuth2TokenEPUrl() {
             if (StringUtils.isBlank(oauth2TokenEPUrl)) {
                 oauth2TokenEPUrl = IdentityUtil.getServerURL("oauth2/token", true, false);
             }
-            return oauth2TokenEPUrl;
+            return IdentityUtil.resolveURL(oauth2TokenEPUrl, true, false);
         }
 
-        public static String getOAuth2DCREPUrl(String tenantDomain) throws URISyntaxException {
```

Comment:
Changing the public method all of a sudden is not good practice as it might break any code that is relying on this particular Util class. 

Best way to introduce this change would be to introduce a new method, deprecate the old method and point to the newly introduced method for future usage.

Then, in a major version of the component, we can remove the deprecated method.

Reply:
Also we can change the old method's implementation to use the newly introduced method

Reply:
> Also we can change the old method's implementation to use the newly introduced method

Agreed on introducing a new method.
With the URL resolver service, DefaultURLResolverService.class takes the tenant name from the ThreadLocal when qualifying paths for tenants. Therefore, the new method would not require the tenant domain as a parameter. Hence, wouldn't it be incorrect if we change the old method to consume the new method?

Reply:
Yes. Let's leave the old method as it is and a comment to use the new method.